File: user-service/src/main/java/com/userservice/infrastructure/apdapters/output/persistence/entity/UserEntity.java
package com.userservice.infrastructure.apdapters.output.persistence.entity;

import jakarta.persistence.*;
import lombok.Builder;

import java.time.LocalDateTime;

@Entity
@Builder
public class UserEntity {

    @Id
    @Column(name = "user_id", length = 36)
    private String id;

    @Column(name = "user_name", length = 10)
    private String userName;

    @Column(name = "email", unique = true, length = 20)
    private String email;

    @Column(name = "password")
    private String password;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "status", length = 10)
    private String status;
}

File: post-service/src/main/kotlin/socat/postservice/infrastructure/persistence/user/entity/UserEntity.kt
package socat.postservice.infrastructure.persistence.user.entity

import jakarta.persistence.*

@Entity
@Table(name = "post_users")
class UserEntity(
    @Id @Column(name = "user_id", length = 36)
    val id: String,

    @Column(name = "user_name", length = 10)
    var userName: String? = null,
)